apps/prairielearn/elements/pl-string-input/pl-string-input.mustache

```diff
@@ -1,28 +1,77 @@
 {{#question}}
+{{#multiline}}
+<script>
+    $(function(){
+        document.querySelectorAll('.multiline-input').forEach((input) => {
+            input.addEventListener('input', function () {
+            // Adjusts the height based on the feedback content. If the feedback changes, the height
+            // changes as well. This is done by resetting the height (so the scrollHeight is computed
+            // based on the minimum height) and then using the scrollHeight plus padding as the new height.
+            this.style.height = '';
+            if (this.scrollHeight) {
+                const style = window.getComputedStyle(this);
+                this.style.height =
+                this.scrollHeight + parseFloat(style.paddingTop) + parseFloat(style.paddingBottom) + 'px';
+            }
+            });
+            input.dispatchEvent(new Event('input'));
+        });
```

Comment:
This function will run once per element, so if someone includes this element multiple times, we'll attach multiple listeners per element. If someone is using this element with external grading, this JS also won't work properly. It's only rendered for the question panel, which means that it won't render (and thus won't be run) when external grader results are delivered via websocket.

My recommendation would be to unconditionally render the script, and also to select the input based on an `id` that contains a UUID. We do this in other elements.

Reply:
https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/c07f73fd59320dc6cb164bb9863166cb4c11a261

Reply:
Changed again in the latest commit: https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/75ebbf7bea3a6a76b7c1d29572bd6f76e4941532

I had a slight issue with this, in that the resizing listener was being used by the other multiline displays in the answer / submission panels, so adding the `id` with a uuid to the selector caused this to only be run for the element on the answer panel with the id. Is there a cleaner way of setting up this listener? Can discuss more on slack.

Reply:
Why not add a consistent ID to all textareas?

Reply:
That would be cleaner, for some reason I thought the IDs had to be unique.

Reply:
They do indeed have to be unique. By "consistent", I mean _add an ID consistently_ to all `textarea` elements so that this script can target them.

Reply:
Ahhh I see what you mean now, the uuid guarantees uniqueness no matter what, that's what was tripping me up.

Reply:
Done here: https://github.com/PrairieLearn/PrairieLearn/pull/9438/commits/ef0feec6580232b9b31287c56b746ecc1955d1a7